### ISSUE TITLE
Add additional safety check for duplice user deletion

### ISSUE THREAD

**Issue by mrIncompetent** [kubermatic/kubermatic#3322]
**User Story**
Duplicated users might lead to project deletion during migration.
The current migration logic only checks if a user has projects via the users project list `user.spec.projects`. To add an additional level of safety we should also check if there is a Project which has a OwnerRef pointing to the User - Otherwise we might delete a user which just lacks `user.spec.projects`

**Comment by mrIncompetent** [comment]
cc @p0lyn0mial

**Comment by mrIncompetent** [comment]
A customer got hit by this.
They manually tried to migrate a project but missed updating the OwnerRef...

Thats the second time we get into trouble with the OwnerRef, shall we rather move away from it?
The garbage collector does not show mercy :/

**Comment by p0lyn0mial** [comment]
I'll look into this. If I recall correctly `user.spec.projects` is not used anymore, was replaced by `UserProjectBindings`.

`OwnerReferences` have nice properties too - automatic deletion :). I would recommend not to manually update resources and doing a mandatory back up before upgrading to a new version.

**Comment by p0lyn0mial** [comment]
I did a small research and here is what I have found.

The `user.spec.projects` field was used only at the beginning and was deprecated soon [after](
https://github.com/kubermatic/kubermatic/blob/master/api/pkg/crd/kubermatic/v1/user.go#L32)

This is also reflected in the migration tool at [phase 6](https://github.com/kubermatic/kubermatic/blob/master/api/cmd/projects-migrator/main.go#L148) where for each entry in the `user.spec.projects` list we create `UserProjectBindings`. At the end the list is [emptied](https://github.com/kubermatic/kubermatic/blob/master/api/cmd/projects-migrator/main.go#L229)

Before a user is deleted the tool checks if the user doesn't belong to a project. It first checks `user.spec.projects` list and if it is empty it also checks `UserProjectBindings`. 

Under normal conditions, the RBAC controller generates `UserProjectBindings` for project's owners (it examines `OnwerReferences`).

I'm not sure what the customer did but a faulty controller might have not created bindings. So as you suggested I'll add an additional check to the tool.
